### ISSUE TITLE
allow optimizer to be stopped from outside

### ISSUE THREAD

**Issue by lelandjobson** [EmilPoulsen/Frixel#19]
- [ ] Move async of optimizer into its own class (can't be wrapped in async)
- [ ] Method that raises a flag to stop optimizer on next iteration

**Comment by EmilPoulsen** [comment]
Yeah, that makes sense. Basically start the optimization from inside the optimizer on a separate thread and then raise events on completion etc.

That way we should be able to stop the process respectfully from outside, without having to kill the whole thread
